include/pfs/parsers.hpp

```diff
@@ -101,28 +106,36 @@ class file_parser
             std::string key;
             std::string value;
             std::tie(key, value) = utils::split_once(line, _delim);
-            if (key.empty())
+
+            std::string remapped_key = key;
+            if (_key_remap)
+            {
+                remapped_key = _key_remap(key);
+            }
+
+            if (remapped_key.empty())
             {
                 throw parser_error("Corrupted line - Missing key", line);
             }
 
             utils::rtrim(key);
+            utils::rtrim(remapped_key);
 
             // Value MIGHT be an empty value, for example:
             // Process without any groups
 
             // If caller specified a set of keys, get only those
-            if (!keys.empty() && keys.find(key) == keys.end())
+            if (!keys.empty() && keys.find(remapped_key) == keys.end())
             {
                 continue;
             }
 
-            auto iter = _parsers.find(key);
+            auto iter = _parsers.find(remapped_key);
             if (iter != _parsers.end())
             {
                 utils::ltrim(value);
                 auto& parser = iter->second;
-                parser(value, output);
+                parser(key, value, output);
```

Comment:
Sorry for throwing another idea here, but I think we can improve further...
Consider the following idea. If the key remap was the following one:
```
void cpu_remap(std::string& key)
{
    static const std::string CPU("cpu");
    if (key.size() == CPU.size() + 1 &&
        key.compare(CPU, 0, CPU.size()) == 0 &&)
    {
        key[CPU.size()] = 'X';
    }
}
```
That way, all specific CPU entries like "cpu1", "cpu2", etc are mapped to "cpuX" for which you can have a separate handler from the total "cpu" entry.
This way you don't need to pass key, and thus avoiding the extra copy per key. WDYT?

Reply:
I think it's OK to pass the key, that way we can give flexibility for other cases in the future, don't you think it's better than just handling this specific case?

Reply:
I agree that this gives flexibility, but on the other hand this copies every single key.
That is fine if there's an actual use case, but at the moment we lose performance and simplicity (all the callbacks receive an unused extra parameter) over nothing...
I think we should treat it as YAGNI till we do. Hope you don't object strongly to my opinion here.